hash.c

```diff
@@ -4802,14 +4825,18 @@ rb_hash_compact(VALUE hash)
 
 /*
  *  call-seq:
- *     hsh.compact! -> hsh or nil
- *
- *  Removes all nil values from the hash.
- *  Returns nil if no changes were made, otherwise returns the hash.
+ *    hash.compact! -> self or nil
  *
- *     h = { a: 1, b: false, c: nil }
- *     h.compact!     #=> { a: 1, b: false }
+ *  Returns +self+ with all +nil+-valued entries removed:
```

Comment:
I think this should have a note that it modifies the object in-place to match the `h1.equal?(h)` example line